[non-strict export] uncovered cases of select and slice

avikchaudhuri opened the pull request:
Summary:
`None` and `Ellipsis` in multi-dimensional indexing was previously not covered.

Moreover, we introduce a small optimization for `slice(None)` and a passthrough when symints do not appear in the indexing.

The remaining case is where indexing is by tensor, which is fairly complicated; we passthrough in that case.

Test Plan:
added tests

Rollback Plan:

Differential Revision: D77943929

pianpwk commented on torch/_export/non_strict_utils.py:
```diff
@@ -983,34 +983,64 @@ def _override(self, func, args, kwargs):
                 if isinstance(item, slice):
+                    if all(s is None for s in (item.start, item.stop, item.step)):
+                        # no-op
+                        return dim + 1, (lambda t: t, [])
```
this case should be covered by https://github.com/pytorch/pytorch/pull/157528

avikchaudhuri commented on torch/_export/non_strict_utils.py:
```diff
@@ -983,34 +983,64 @@ def _override(self, func, args, kwargs):
+                    n_none_slices -= 1
+
+            # only rewrite when there are symints
+            if has_symint:
```
Yeah it should pass. Which is why I have the `G_*` modules which I used to test concrete inputs locally. 

I just didn't want to perturb existing generated code too much. In a separate PR I had tried doing this for strict and had to update a gazillion tests.

avikchaudhuri committed:
[non-strict export] uncovered cases of select and slice (#157821)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/157821

`None` and `Ellipsis` in multi-dimensional indexing was previously not covered.

Moreover, we introduce a small optimization for `slice(None)` and a passthrough when symints do not appear in the indexing.

The remaining case is where indexing is by tensor, which is fairly complicated; we passthrough in that case.

Test Plan:
added tests

Rollback Plan:

Reviewed By: pianpwk

Differential Revision: D77943929
